packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -299,9 +299,13 @@ export const getL1ToL2MessageDataFromL1TxHash = async ({
     if (isEthDeposit) {
       // nitro eth deposit
       const [ethDepositMessage] = await l1TxReceipt.getEthDeposits(l2Provider)
+      // when depositing to a custom address
+      const [l1ToL2Message] = await l1TxReceipt.getL1ToL2Messages(l2Provider)
```

Comment:
The way I see it we can't pass a prop because we are calling the methods to find out what type of tx we are facing.

However, because the majority of deposits will be to a non-custom destination address, I optimized it so that we return the object after `getEthDeposits` is successful. This way we only call  `l1ToL2Message` if the former is missing - which should be rare.